---
title: "RNAseq Differential Expression Fox River Mixture"
output:
  pdf_document: default
  html_notebook: default
---
```{r}
BiocManager::install("Mus.musculus")
BiocManager::install("enrichplot")
BiocManager::install("clusterProfiler")
BiocManager::install("scRNAseq")
BiocManager::install("SingleCellExperiment")
BiocManager::install("TOAST")
devtools::install_github('xuranw/MuSiC')
remotes::install_github("renozao/xbioc")

# install the MuSiC2 package
if (!"MuSiC2" %in% rownames(installed.packages())) {
  devtools::install_github('Jiaxin-Fan/MuSiC2')
}
```


The majority of Library packages are listed here. If you want to include more just install and load at the end of the list. If code isn't recognizing the function, it is likely that R didn't load the package that the function comes from.
#load libraries every time
```{r, results='hide'}
library(ggplot2)
library(devtools)
library(rgl)
library(tidyverse)
library(tidyr)
library(RColorBrewer)
library(expss)
library(dplyr)
library(BiocManager)
library(RMariaDB)
library(GenomicAlignments)
library(BiocParallel)
library(GenomicFeatures)
library(DESeq2)
library(Mus.musculus)
library(Rsamtools)
library(pheatmap)
library(topGO)
library(gridExtra)
library(AnnotationDbi)
library(org.Mm.eg.db)
library(EnhancedVolcano)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(enrichplot)
library(clusterProfiler)
library(pathview)
library(readr)
library(MuSiC2)
library(ggpubr)
library(ggsignif)
library(rstatix)
library(scRNAseq)
library(VennDiagram)
library(stringr)
library(tibble)
```

Set your working directory and pull in your reference genome to compare your samples to. We will read the gene model from an Ensembl GTF file. Be sure your gene model reflects the same model used in your samples. 

***Define the Gene Model***
```{r}
TxDb<-makeTxDbFromUCSC(genome = "mm10",
                        tablename ="knownGene", 
                        goldenPath.url = getOption("UCSC.goldenPath.url"))
```

For additional notes reference "https://www.bioconductor.org/help/course-materials/2016/CSAMA/lab-3-rnaseq/rnaseq_gene_CSAMA2016.html"

The following line produces a GRangesList of all the exons grouped by gene (Lawrence et al. 2013). Each element of the list is a GRanges object of the exons for a gene
```{r}
ebg <- exonsBy(TxDb, 
               by="gene")
ebg
g_ids<-names(ebg)
```

Now that our gene model reference is ready we can load in samples and analyze them accordingly. I found separating by tissue type was easiest. But it can also be helpful to load all samples together. 

***Liver Samples***
```{r}
info<- read.csv(file = "RNA-seq_metadata.csv", #reading in the sample information
                   header = T, 
                   sep = ",")

filenames <- file.path("P:/OEH/Faculty/hlehmler/research/Lehmler Group-Amanda Bullert/RNAseq/fastq_files/BAM_files/", 
                           info$bam_id)
file.exists(filenames) #a sanity check to make sure you have files named
```

Next specify the details about how the BAM files should be treated in R, e.g., only process 2 million reads at a time. This can be modified to your computing limitations.
```{r}
bamfiles <- BamFileList(filenames, 
                            yieldSize=2000000)
seqinfo(bamfiles)
```

#***Counting***
```{r}

se <- summarizeOverlaps(features=ebg, #define the gene reference
                            reads=bamfiles, #samples to be read
                            mode="Union", 
                            singleEnd=FALSE, #False indicates samples are paired-end
                            ignore.strand=FALSE, #not a strand specific experiment
                            BPPARAM= SerialParam(progressbar = TRUE)) #progress bar shown
se
head(assay(se)) #access the counts
str(metadata(rowRanges(se))) #just to look a the structure of data

```


#proceed here (no need to recount summarized experiments)

```{r}
colData(se) #metadata about the samples
colData(se)<-DataFrame(info) #take the sample info and assign it as the metadata
se$Treatment<- as.factor(se$Treatment) #organizing structure of groups
se$Treatment1<- relevel(se$Treatment, 
                           "WT_VEH") # tells the system which group is "control"
se$Treatment2<- relevel(se$Treatment, 
                           "WT_PCB6") # tells the system which group is "control"
se$Treatment3<- relevel(se$Treatment, 
                           "WT_PCB30") # tells the system which group is "control"
se$Treatment4<- relevel(se$Treatment, 
                           "GF_VEH") # tells the system which group is "control"
se$Treatment5<- relevel(se$Treatment, 
                           "GF_PCB6") # tells the system which group is "control"
se <- se[ rowSums(assay(se)) >= 10, ] #remove genes that have a total count less than 10, a good prefilter measure
```

#***Using DeSeq2***
We can now construct a DESeqDataSet object to formulate a starting point for our analysis. You need to add an appropriate design for analysis
```{r}
dds <- DESeqDataSet(se, 
                    design = ~ Treatment1) 

dds2 <- DESeqDataSet(se, 
                    design = ~ Treatment2) 

dds3 <- DESeqDataSet(se, 
                    design = ~ Treatment3) 

dds4 <- DESeqDataSet(se, 
                    design = ~ Treatment4) 

dds5 <- DESeqDataSet(se, 
                    design = ~ Treatment5) 
```

#***Exploratory analysis and visualization***
*transformations*
The variance stabilizing transformation (VST) a goal of stablizing the variance across the range of values. produce log2-like values for high counts. 
```{r}
vsd <- vst(dds)
```

#results
#***Differential Expression***
Lets run a differential expression pipeline using the raw counts

This function will print out a message for the various steps it performs. These are described in more detail in the manual page for DESeq, which can be accessed by typing ?DESeq. Briefly these are: the estimation of size factors (controlling for differences in the sequencing depth of the samples), the estimation of dispersion values for each gene, and fitting a generalized linear model.
```{r}
dds <- DESeq(dds)
dds2 <- DESeq(dds2)
dds3 <- DESeq(dds3)
dds4 <- DESeq(dds4)
dds5 <- DESeq(dds5)

resultsNames(dds2)
GV_WTV<-results(dds, name="Treatment1_GF_VEH_vs_WT_VEH")
WT6_WTV<-results(dds, name = "Treatment1_WT_PCB6_vs_WT_VEH")
WT30_WTV<-results(dds, name = "Treatment1_WT_PCB30_vs_WT_VEH")
G6_WT6<-results(dds2, name="Treatment2_GF_PCB6_vs_WT_PCB6")
WT30_WT6<-results(dds2, name="Treatment2_WT_PCB30_vs_WT_PCB6")
G30_WT30<-results(dds3, name="Treatment3_GF_PCB30_vs_WT_PCB30")
G30_GV<-results(dds4, name="Treatment4_GF_PCB30_vs_GF_VEH")
G6_GV<-results(dds4, name="Treatment4_GF_PCB6_vs_GF_VEH")
G30_G6<-results(dds5, name="Treatment5_GF_PCB30_vs_GF_PCB6")

```
Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable in the design formula. If there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level. Treatment 6 vs vehicle
```{r}
summary(GV_WTV)
summary(WT6_WTV)
summary(WT30_WTV)
summary(G6_WT6)
summary(WT30_WT6)
summary(G30_WT30)
summary(G30_GV)
summary(G6_GV)
summary(G30_G6)
```

```{r}
res1<-GV_WTV
res2<-WT6_WTV
res3<-WT30_WTV
res4<-G6_WT6
res5<-WT30_WT6
res6<-G30_WT30
res7<-G30_GV
res8<-G6_GV
res9<-G30_G6
```

#***Annotating results***

How to assign actual gene names to our counts. Using an annotation package for Mus musculus. 
```{r}
res1$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res1), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res2$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res2), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res3$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res3), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res4$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res4), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res5$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res5), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res6$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res6), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res7$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res7), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res8$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res8), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res9$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res9), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
res1$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res1), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res2$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res2), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res3$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res3), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res4$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res4), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res5$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res5), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res6$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res6), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res7$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res7), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res8$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res8), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res9$entrez <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res9), #using rownames
                     column="ENTREZID", #add a column of entrezs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
res1$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res1), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res2$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res2), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res3$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res3), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res4$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res4), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res5$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res5), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res6$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res6), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res7$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res7), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res8$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res8), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res9$GENENAME <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res9), #using rownames
                     column="GENENAME", #add a column of GENENAMEs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
res1$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res1), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res2$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res2), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res3$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res3), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res4$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res4), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res5$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res5), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res6$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res6), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res7$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res7), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res8$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res8), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res9$GO <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res9), #using rownames
                     column="GO", #add a column of GOs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
res1$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res1), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res2$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res2), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res3$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res3), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res4$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res4), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res5$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res5), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res6$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res6), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res7$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res7), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res8$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res8), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
res9$ENSEMBL <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(res9), #using rownames
                     column="ENSEMBL", #add a column of ENSEMBLs associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

res1<-GV_WTV
res2<-WT6_WTV
res3<-WT30_WTV
res4<-G6_WT6
res5<-WT30_WT6
res6<-G30_WT30
res7<-G30_GV
res8<-G6_GV
res9<-G30_G6


```{r}
write.csv( as.data.frame(res1), file="Results_spreadsheets/GF_veh_vs_WT_VEH_results.csv" )
write.csv( as.data.frame(res2), file="Results_spreadsheets/WT_PCB6_vs_WT_VEH_results.csv" )
write.csv( as.data.frame(res3), file="Results_spreadsheets/WT_PCB30_vs_WT_VEH_results.csv" )
write.csv( as.data.frame(res4), file="Results_spreadsheets/GF_PCB6_vs_WT_PCB6_results.csv" )
write.csv( as.data.frame(res5), file="Results_spreadsheets/WT_PCB30_vs_WT_PCB6_results.csv" )
write.csv( as.data.frame(res6), file="Results_spreadsheets/GF_PCB30_vs_WT_PCB30_results.csv" )
write.csv( as.data.frame(res7), file="Results_spreadsheets/GF_PCB30_vs_GF_VEH_results.csv" )
write.csv( as.data.frame(res8), file="Results_spreadsheets/GF_PCB6_vs_GF_VEH_results.csv" )
write.csv( as.data.frame(res9), file="Results_spreadsheets/GF_PCB30_vs_GF_PCB6_results.csv" )
```


```{r}
resOrdered1 <- res1[order(res1$padj),] #reorder the genes based on significance
resOrdered2 <- res2[order(res2$padj),] #reorder the genes based on significance
resOrdered3 <- res3[order(res3$padj),] #reorder the genes based on significance
resOrdered4 <- res4[order(res4$padj),] #reorder the genes based on significance
resOrdered5 <- res5[order(res5$padj),] #reorder the genes based on significance
resOrdered6 <- res6[order(res6$padj),] #reorder the genes based on significance
resOrdered7 <- res7[order(res7$padj),] #reorder the genes based on significance
resOrdered8 <- res8[order(res8$padj),] #reorder the genes based on significance
resOrdered9 <- res9[order(res9$padj),] #reorder the genes based on significance
```


```{r}
my_list<-list( "Germ Free Vehicle vs Wild Type Vehicle"= res1,
              "Wild Type 6 mg vs Wild Type Vehicle"= res2,
              "Wild Type 30 mg vs Wild Type Vehicle"= res3,
              "Germ Free 6 mg vs Wild Type 6 mg"= res4,
              "Wild Type 30 mg vs Wild Type 6 mg"= res5,
              "Germ Free 30 mg vs Wild Type 30 mg"= res6,
              "Germ Free 30 mg vs Germ Free Vehicle" = res7,
              "Germ Free 6 mg vs Germ Free Vehicle" = res8,
              "Germ Free 30 mg vs Germ Free 6 mg" = res9)

```

#for loop for Gene and kegg enrichment
#gene enrichment
```{r}

plot_list_enrich<- list()
for (i in names(my_list)) {
    print(i)
  x<- my_list[[i]][order(my_list[[i]]$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ENSEMBL

gse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")

dot<-dotplot(gse, showCategory=10, split=".sign") +
  facet_grid(.~.sign)+
  ggtitle(paste(i,"Gene Enrichment"))+ theme(axis.text.y = element_text(size = 5))    

plot_list_enrich[[i]] <- dot

print(dot)

  data<-data.frame(gse)
    write.csv(data ,file=paste('gene_enrichment/',i,'gene enrichment.csv'))
}

```

```{r}
for (i in names(plot_list_enrich)) {
tiff(file = paste('gene_enrichment/',i,'gene enrichment.tiff'), units="in", width=7, height=5, res=1000)
print(plot_list_enrich[[i]])
dev.off()
}  
```


#KEGG pathview

```{r}

plot_list_kegg<- list()
for (i in names(my_list)) {
    print(i)
  x<- my_list[[i]][order(my_list[[i]]$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez

kegg_organism = "mmu"
kk3 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

dot<-dotplot(kk3, showCategory=10, split=".sign") +
  facet_grid(.~.sign)+
  ggtitle(paste(i,"KEGG Enrichment"))+ theme(axis.text.y = element_text(size = 5))    

plot_list_kegg[[i]] <- dot

print(dot)

  data<-data.frame(kk3)
    write.csv(data ,file=paste('KEGG_enrichment/',i,'KEGG enrichment.csv'))
}

```

```{r}
for (i in names(plot_list_kegg)) {
tiff(file = paste('KEGG_enrichment/',i,'KEGG enrichment.tiff'), units="in", width=7, height=5, res=1000)
print(plot_list_kegg[[i]])
dev.off()
}  
```


# Produce the native KEGG plot (PNG)
```{r}
# Produce the native KEGG plot (PNG)
for (i in names(my_list)) {
  x<- my_list[[i]][order(my_list[[i]]$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
  
z <- pathview(gene.data=gene_list, 
                          pathway.id="00565", 
                          species = kegg_organism, 
                          out.suffix = i)
zz<-data.frame(z$plot.data.gene)
write.csv(zz, file = "lipid metabolism/kegg_to_gene_id.csv")
}
```

```{r}
# Produce the native KEGG plot (PNG)
for (i in names(my_list)) {
  x<- my_list[[i]][order(my_list[[i]]$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
  
z <- pathview(gene.data=gene_list, 
                          pathway.id="00564", 
                          species = kegg_organism, 
                          out.suffix = i)

zz<-data.frame(z$plot.data.gene)
write.csv(zz, file = "glycerophospholipid metabolism/kegg_to_gene_id.csv")
}
```

#plotting individual genes of interest (Liver)
Using normalized counts (prebuilt into DESEQ2) which normalizes counts by the estimated size factors (or normalization factors if these were used) and adds a pseudocount of 1/2 to allow for log scale plotting.
**drug metabolizing enzymes**
```{r}
top_gene<-read.csv("gene_list_drug_metabolizing_kegg.csv", header = FALSE)
top_gene<-as.character(top_gene$V1)

setdiff(top_gene,rownames(counts(dds))) #check if all genes of interest can be found in the rownames of the deseq object --> if not you will not be able to run the block of code. Character(0) or integer(0) is a what you want to see
```

```{r}
stat.test<-read.csv("FDR.csv")
stat.test$symbol<-factor(stat.test$symbol)
stat.test$group2<-as.factor(stat.test$group2)
stat.test$group1<-as.factor(stat.test$group1)
stat.test
```

```{r}
data_1 <- split(as.tibble(stat.test[,2:4]), as.factor(stat.test$symbol))
```
```{r}
dds$Treatment<-factor(dds$Treatment, levels = c("WT_VEH","WT_PCB6","WT_PCB30","GF_VEH","GF_PCB6","GF_PCB30"))
```


```{r message=FALSE}
plot_list<-list()
for (i in unique(1:length(top_gene))){
  gene <- top_gene[i]
  b<- plotCounts(dds, gene = gene, intgroup = "Treatment",normalized = TRUE, returnData = TRUE)
  z = max(b$count) + (max(b$count)*0.05)
  d <- ggplot(b, aes(factor(Treatment), count))+
    geom_violin(mapping = aes(x = Treatment, 
                              y = count, 
                              fill = Treatment))+
    #adding jitter to avoid overplotting
    geom_point(mapping = aes(x = Treatment, 
                             y = count, 
                             fill = Treatment, 
                             shape= Treatment), 
               size = 5, position = position_jitter(width = 0.3, height=0))+ 
    scale_y_continuous(expand = c(0,0) , limits = c(0,z+z*0.4)) +
    stat_summary(mapping = aes(x = Treatment, y = count), 
                 geom = 'crossbar', 
                 fun= 'mean', 
                 colour = 'black', 
                 width=0.2)+
  scale_fill_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF","#666666","#FF66CC")) +
  scale_shape_manual(values = c(19, 15, 17, 18,25,16)) +
  labs(x = NULL, y = "Normalized counts")+ # changes the y axis label, removes x axis label
  theme_classic(base_size = 20)+# changes the overall style of the plot
  ggtitle(paste0(res1$symbol[gene])) +
  theme(
    axis.text.x = element_text(colour = "black", size = 10, angle = 15),
    axis.text.y = element_text(colour = "black"))+
  stat_pvalue_manual(data_1[[i]], 
    y.position = z, step.increase = 0.2,
    label = "p = {scales::pvalue(p.adj)}", hide.ns = TRUE
    )
  plot_list[[gene]] <- d
}

head(plot_list)
```


```{r warning=FALSE}
# Export into pdf: display multiple plots on the same page
ggexport(
  plotlist = plot_list, filename = "gene_plot.pdf", 
  ncol = 1, nrow = 1, height = 7, width = 7, res = 600,pointsize = 8
)
```

